test/test_linearizer.py

```diff
@@ -1420,6 +1420,64 @@ def test_padto_where(self):
       [Opt(OptOps.PADTO, 0, 32), Opt(OptOps.UPCAST, 0, 8),],
     ])
 
+  @unittest.skip("multireduce isn't supported yet")
+  def test_padto_sum_multireduce(self):
+    N = 18 * 18
+    x = (Tensor.rand(N, N).shrink(((0, 17), (0, 17)))*100).realize()
+    x_ast = LazyOp(op=BinaryOps.ADD, src=(LazyOp(op=BufferOps.LOAD, src=(), arg=MemBuffer(idx=1, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(17, 17), strides=(324, 1), offset=0, mask=None, contiguous=False),)))),LazyOp(op=BufferOps.CONST, src=(), arg=ConstBuffer(val=100.0, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(1, 1), strides=(0, 0), offset=0, mask=None, contiguous=False),)))))) # noqa: E501
+    def ast(axis:int): LazyOp(op=BufferOps.STORE, src=(LazyOp(op=ReduceOps.SUM, src=(LazyOp(op=BinaryOps.SUB, src=(x_ast,LazyOp(op=ReduceOps.SUM, src=(x_ast,), arg=(axis,)))),), arg=(axis,)),), arg=MemBuffer(idx=0, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(1, 17), strides=(0, 1), offset=0, mask=None, contiguous=True),)))) # noqa: E501
+    opts = [[Opt(OptOps.PADTO, 0, 32)],[Opt(OptOps.PADTO, 0, 32), Opt(OptOps.UPCAST, 0, 8),],]
+    helper_linearizer_ast((ast(0),), [x], opts=opts)
+    helper_linearizer_ast((ast(1),), [x], opts=opts)
+
+    # pad reduce sum
+    helper_linearizer_ast((ast(0),), [x], opts=[[Opt(OptOps.PADTO, 1, 32)],])
+    ast = LazyOp(op=BufferOps.STORE, src=(LazyOp(op=ReduceOps.SUM, src=(LazyOp(op=BinaryOps.SUB, src=(x_ast,LazyOp(op=ReduceOps.SUM, src=(x_ast,), arg=(0,1)))),), arg=(0,1)),), arg=MemBuffer(idx=0, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(1, 1), strides=(0, 1), offset=0, mask=None, contiguous=True),)))) # noqa: E501
+    helper_linearizer_ast((ast,), [x], opts=[[Opt(OptOps.PADTO, 0, 32)],])
+
+  @unittest.skip("multireduce isn't supported yet")
+  def test_padto_max_multireduce(self):
```

Comment:
I'm not sure if this is needed because you can't pad the reduce axis anyways; but I included it for posterity